app/medInria/medRegistrationWorkspace.cpp

```diff
@@ -178,37 +167,32 @@ void medRegistrationWorkspace::updateFromMovingContainer()
         d->movingLayerGroup->addImpactedlayer(movingView, movingData);
         d->movingLayerGroup->addImpactedlayer(fuseView, movingData);
     }
-    if (!d->selectorToolBox->setMovingData(movingData))
-    {
-        // delete the view because something failed at some point
-        d->viewGroup->removeImpactedView(movingView);
-        d->movingLayerGroup->removeImpactedlayer(movingView, movingData);
-        d->movingLayerGroup->removeImpactedlayer(fuseView, movingData);
-        movingView->deleteLater();
-    }
+
+    toolbox->setMovingData(movingData);
 }
 
 void medRegistrationWorkspace::updateFromFixedContainer()
 {
```

Comment:
In fact, `updateFromMovingContainer` and `updateFromFixedContainer` are really similar and can be factorized :)